packages/xsnap/makefiles/lin/xsnap.mk

```diff
@@ -0,0 +1,139 @@
+% : %.c
+%.o : %.c
+
+GOAL ?= debug
+NAME = xsnap
+ifneq ($(VERBOSE),1)
+MAKEFLAGS += --silent
+endif
+
+# This Makefile gets invoked as "make" with this directory as the "cwd" using
+# Node.js child_process.spawn. For reasons that remain mysterious to this
+# author, under these circumstances, $(PWD) is two parent directories up
+# from this file. At the shell, (cd makefiles/lin && make) works fine.
+# Regardless, using HERE instead of PWD makes this Makefile work regardless of
+# the working directory.
+HERE = $(dir $(realpath $(firstword $(MAKEFILE_LIST))))
+MODDABLE = $(HERE)/../../moddable
+XS_DIR = $(MODDABLE)/xs
```

Comment:
Can/should we put a check in here somewhere to test that the $MODDABLE directory is present? I'm thinking about folks who have had a tree checked out for a while, and don't have the git submodule in place, and what sort of confusion this Makefile would suffer if the XS sources are missing.

Reply:
There’s a build rule down from here that brings the submodule into place if it hasn’t been already. Without that, this does indeed break spectacularly, but not too mysteriously.